### PR TITLE
[DNM] On demand scheduling

### PR DESCRIPTION
**Pull request by antonva**

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2601159

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2633063

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2638901

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2640062

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2679175

File: runtime/polkadot/src/lib.rs [hunk 1]
```diff
@@ -1373,6 +1373,7 @@ construct_runtime! {
 		Hrmp: parachains_hrmp::{Pallet, Call, Storage, Event<T>, Config} = 60,
 		ParaSessionInfo: parachains_session_info::{Pallet, Storage} = 61,
 		ParasDisputes: parachains_disputes::{Pallet, Call, Storage, Event<T>} = 62,
+		ParaAssignmentProvider: parachains_assigner_parachains::{Pallet} = 63,
```
[the remark this answers is not in the document]
Comment (antonva) [reply runtime/polkadot/src/lib.rs:1376]: This is the parachains assignment pallet, not parathreads, the proposed changes to the scheduler pallet won't work without an `AssignmentProvider`. 

In order to be able to merge to master we might want to maintain both the old and new scheduler as separate modules that could then be swapped out. There are some changes I'm aware of in the `paras*` modules that might need to be duplicated for the short term. cc @alexgparity

**Pull request closed without merging**